### PR TITLE
docs: update reference architectures heat map and key management

### PR DESCRIPTION
**Pull request by SirSpidey**
### Description

- Heat map: Vincent: Remove "Enterprise connectivity" and "BYOIF/Edge gateway" as those are not technically included in the automation."
- ~~Replace Secrets Manager with generic key management~~
    - Remove the `Key and password management architecture decisions` table from the VSI RAs based on your earlier feedback.
- Update the version in the reference architectures (RA) to match the latest in catalog. The versions are displayed in the Cloud docs.
- Replace the diagrams that represent the patterns with the "single-sourced" svgs in the `reference-architectures` directory. Some .png files remain that describe parts of architectures.

#### No release required

- [ ] Examples or tests (addition or updates of examples or tests)
- [x] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [ ] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [ ] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

Replace this text with information that users need to know about the bug fixes, features, and breaking changes. This information helps the merger write the commit message that is published in the release notes for the module.

---

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

File: reference-architectures/deploy-arch-ibm-slz-ocp.md [hunk 1]
```diff
@@ -100,7 +100,7 @@ business challenge, or target cloud environments.
...
+| * Use public and private SSH keys to access virtual server instances by using SSH  \n * Use SSH proxy to log in to all virtual server instances by using the bastion host  \n * Do not store private SSH key on any virtual instances, also not on the bastion host  \n * Do not allow any other SSH login methods except the one with the specified public and private SSH key pair | Public and private SSH keys provided by customer | Ask customer to specify the keys. Accept the input as secure parameter or as reference to the key stored in key management. Do not print SSH keys in any log files. Do not persist private SSH key. | |
```
Comment (ocofaigh) [inline reference-architectures/deploy-arch-ibm-slz-ocp.md:103]: OCP DA doesn't have virtual servers. It also doesn't require SSH key input either

File: reference-architectures/deploy-arch-ibm-slz-vsi-quickstart.md [hunk 2]
```diff
@@ -96,5 +96,5 @@ business challenge, or target cloud environments.
 
 | Requirement | Component | Reasons for choice | Alternative choice |
 |-------------|-----------|--------------------|--------------------|
-| * Use public and private SSH keys to access virtual server instances by using SSH  \n * Do not store private SSH key on any virtual instances, also not on the jump box host  \n * Do not allow any other SSH login methods except the one with the specified public and private SSH key pair | Public and private SSH keys provided by customer | Ask customer to specify the keys. Accept the input as secure parameter or as reference to the key stored in IBM Cloud Secrets Manager. | |
+| * Use public and private SSH keys to access virtual server instances by using SSH  \n * Do not store private SSH key on any virtual instances, also not on the jump box host  \n * Do not allow any other SSH login methods except the one with the specified public and private SSH key pair | Public and private SSH keys provided by customer | Ask customer to specify the keys. Accept the input as secure parameter or as reference to the key stored in key management. | |
```
Comment (ocofaigh) [inline reference-architectures/deploy-arch-ibm-slz-vsi-quickstart.md:99]: I'm not sure this statement is correct. There is one required input variable called `ssh_public_key`. I don't think there is any way to "reference to the key stored in key management"? And there is also no input for private SSH keys. Not sure if they are in play here?

File: reference-architectures/deploy-arch-ibm-slz-vsi-standard.md [hunk 3]
```diff
@@ -104,7 +104,7 @@ business challenge, or target cloud environments.
 
 | Requirement | Component | Reasons for choice | Alternative choice |
 |-------------|-----------|--------------------|--------------------|
-| * Use public and private SSH keys to access virtual server instances by using SSH  \n * Do not store private SSH key on any virtual instances  \n * Do not allow any other SSH login methods except the one with the specified public and private SSH key pair | Public and private SSH keys provided by customer | Ask customer to specify the keys. Accept the input as secure parameter or as reference to the key stored in IBM Cloud Secrets Manager. | |
+| * Use public and private SSH keys to access virtual server instances by using SSH  \n * Do not store private SSH key on any virtual instances  \n * Do not allow any other SSH login methods except the one with the specified public and private SSH key pair | Public and private SSH keys provided by customer | Ask customer to specify the keys. Accept the input as secure parameter or as reference to the key stored in key management. | |
```
Comment (ocofaigh) [inline reference-architectures/deploy-arch-ibm-slz-vsi-standard.md:107]: same as above https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/pull/409/files#r1171071185

File: reference-architectures/deploy-arch-ibm-slz-vsi-quickstart.md (hunk 2 above)
Comment (SirSpidey) [reply reference-architectures/deploy-arch-ibm-slz-vsi-quickstart.md]: @ocofaigh I've removed the `Key and password management architecture decisions` table based on this information. It was already removed in the VPC ref arch.

File: reference-architectures/deploy-arch-ibm-slz-vsi-standard.md (hunk 3 above)
Comment (SirSpidey) [reply reference-architectures/deploy-arch-ibm-slz-vsi-standard.md]: I've removed the `Key and password management architecture decisions` table as above.

**Comment by SirSpidey** [comment]
@ocofaigh: several new commits now in this PR

- Update version in the reference architectures (RA) to match the latest in catalog. The versions are displayed in the Cloud docs.
- Remove the `Key and password management architecture decisions` table from the VSI RAs based on your earlier feedback.
- Replace the diagrams that represent the patterns with the "single-sourced" svgs in the `reference-architectures` directory. Some .png files remain that describe parts of architectures.

File: reference-architectures/deploy-arch-ibm-slz-vsi-standard.md [hunk 4]
```diff
@@ -12,7 +12,7 @@ authors:
   - name: "Vincent Burckhardt"
 
 # The release that the reference architecture describes
-version: 3.5.1
+version: 3.5.4
```
Comment (ocofaigh) [inline reference-architectures/deploy-arch-ibm-slz-vsi-standard.md:15]: 3.6.4

File: reference-architectures/deploy-arch-ibm-slz-vsi-standard.md [hunk 5]
```diff
@@ -52,7 +52,7 @@ content-type: reference-architecture
 {: toc-industry="Banking,FinancialSector"}
 {: toc-use-case="Cybersecurity"}
 {: toc-compliance="FedRAMP"}
-{: toc-version="3.5.1"}
+{: toc-version="3.5.4"}
```
Comment (ocofaigh) [inline reference-architectures/deploy-arch-ibm-slz-vsi-standard.md:55]: 3.6.4

File: reference-architectures/deploy-arch-ibm-slz-ocp.md [hunk 6]
```diff
@@ -12,7 +12,7 @@ authors:
   - name: "Vincent Burckhardt"
 
 # The release that the reference architecture describes
-version: 3.5.1
+version: 3.5.4
```
Comment (ocofaigh) [inline reference-architectures/deploy-arch-ibm-slz-ocp.md:15]: 3.6.4

File: reference-architectures/deploy-arch-ibm-slz-ocp.md [hunk 7]
```diff
@@ -49,7 +49,7 @@ content-type: reference-architecture
 {: toc-industry="Banking,FinancialSector"}
 {: toc-use-case="Cybersecurity"}
 {: toc-compliance="FedRAMP"}
-{: toc-version="3.5.1"}
+{: toc-version="3.5.4"}
```
Comment (ocofaigh) [inline reference-architectures/deploy-arch-ibm-slz-ocp.md:52]: 3.6.4

File: reference-architectures/deploy-arch-ibm-slz-vpc-standard.md [hunk 8]
```diff
@@ -12,7 +12,7 @@ authors:
   - name: "Vincent Burckhardt"
 
 # The release that the reference architecture describes
-version: 3.5.1
+version: 3.5.4
```
Comment (ocofaigh) [inline reference-architectures/deploy-arch-ibm-slz-vpc-standard.md:15]: 3.6.4

File: reference-architectures/deploy-arch-ibm-slz-vpc-standard.md [hunk 9]
```diff
@@ -49,7 +49,7 @@ content-type: reference-architecture
 {: toc-industry="Banking,FinancialSector"}
 {: toc-use-case="Cybersecurity"}
 {: toc-compliance="FedRAMP"}
-{: toc-version="3.5.1"}
+{: toc-version="3.5.4"}
```
Comment (ocofaigh) [inline reference-architectures/deploy-arch-ibm-slz-vpc-standard.md:52]: 3.6.4

File: reference-architectures/deploy-arch-ibm-slz-vsi-quickstart.md [hunk 10]
```diff
@@ -12,7 +12,7 @@ authors:
   - name: "Vincent Burckhardt"
 
 # The release that the reference architecture describes
-version: 3.5.1
+version: 3.5.4
```
Comment (ocofaigh) [inline reference-architectures/deploy-arch-ibm-slz-vsi-quickstart.md:15]: 3.6.4

File: reference-architectures/deploy-arch-ibm-slz-vsi-quickstart.md [hunk 11]
```diff
@@ -47,7 +47,7 @@ content-type: reference-architecture
 {: toc-content-type="reference-architecture"}
 {: toc-industry="Banking,FinancialSector"}
 {: toc-use-case="Cybersecurity"}
-{: toc-version="3.5.1"}
+{: toc-version="3.5.4"}
```
Comment (ocofaigh) [inline reference-architectures/deploy-arch-ibm-slz-vsi-quickstart.md:50]: 3.6.4

**Review by ocofaigh — CHANGES_REQUESTED** [review]
version should be 3.6.4

File: reference-architectures/deploy-arch-ibm-slz-vsi-standard.md (hunk 4 above)
Comment (SirSpidey) [reply reference-architectures/deploy-arch-ibm-slz-vsi-standard.md:15]: Sorry about that... 😿

**Review by ocofaigh — APPROVED** [review]

File: README.md [hunk 12]
```diff
@@ -417,7 +417,7 @@ Users can add a name and optionally a public key. If `public_key` is not provide
 
 #### vsi variable
 
-Note - You can't make changes to the VSI image with this module. That restriction is in place so that you don't inadvertently create an outage or lose data.
+Note - You can't change the VSI image with this module. That restriction is in place so that you don't inadvertently create an outage or lose data.
```
Comment (vburckhardt) [inline README.md:420]: I think this should read as "once the infrastructure has been created, any change to this variable won't apply". The current way it reads imply that it is not possible to change the VSI image at all.

**Review by vburckhardt — COMMENTED** [review]
Hi Allen, thanks for this - just one comment.

File: README.md (hunk 12 above)
Comment (SirSpidey) [reply README.md]: @vburckhardt Made a change for your review

image

**Review by vburckhardt — APPROVED** [review]

**Pull request merged**